chain/default_syncer.go

```diff
@@ -208,6 +212,10 @@ func (syncer *DefaultSyncer) syncOne(ctx context.Context, parent, next types.Tip
 		return nil
 	}
 
+	if err := syncer.validator.ValidateSemantics(ctx, next, parent); err != nil {
```

Comment:
If the syncer has a `validator` field `RunStateTransition` should probably be a method on that field too for consistency.